tests/ops/qubit/test_matrix_ops.py

```diff
@@ -669,9 +685,53 @@ def test_unitary(self, input_matrix, wires):
         mat = qml.matrix(qml.BlockEncode(input_matrix, wires))
         assert np.allclose(np.eye(len(mat)), mat.dot(mat.T.conj()))
 
-    # TODO: add tensorflow support
-    # def test_blockencode_tf(self):
-    #     """Test that the BlockEncode operator matrix is correct for tf."""
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "msg"),
+        [
+            (
+                [[0, 1], [1, 0]],
+                1,
+                f"Block encoding a \({2} x {2}\) matrix requires a hilbert space of size"
+                f" at least \({4} x {4}\). Cannot be embedded in a {1} qubit system.",
+            ),
+        ],
+    )
+    def test_correct_error_message(self, input_matrix, wires, msg):
+        with pytest.raises(ValueError, match=msg):
+            op = qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "output_matrix"),
+        [
+            (1.0, 0, [[1, 0], [0, -1]]),
+            (0.3, 0, [[0.3, 0.9539392], [0.9539392, -0.3]]),
+            (
+                [[0.1, 0.2], [0.3, 0.4]],
+                range(2),
+                [
+                    [0.1, 0.2, 0.97283788, -0.05988708],
+                    [0.3, 0.4, -0.05988708, 0.86395228],
+                    [0.94561648, -0.07621992, -0.1, -0.3],
+                    [-0.07621992, 0.89117368, -0.2, -0.4],
+                ],
+            ),
+            (
+                0.1,
+                range(2),
+                [[0.1, 0.99498744, 0, 0], [0.99498744, -0.1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]],
+            ),
+        ],
```

Comment:
Should this be defined for each interface separately? I suppose the tests for different interfaces use the same data?